### PR TITLE
Markdownlint

### PR DESCRIPTION
**Pull request by fiskus**
Installed `markdownlint-cli` and set up `npm run lint:docs` CLI command to lint Markdown docs.
To make changes more manageable and incremental, I disabled lint on almost all files, except `docs/Catalog/Preferences.md` for a sample.

### CODE REVIEW

**Review by nl0 — CHANGES_REQUESTED** [review]
looks kinda useful, but breaks some actions

File: docs/Catalog/Preferences.md [hunk 1]
```diff
@@ -48,16 +48,21 @@ drag-and-drop or from folders in S3
...
+that map to an empty object reserved for future enhancements;
 buckets in this dictionary are the ones offered when the user clicks
-Revise Package > Add files from Bucket; if the dictionary is not set or is empty the feature "Add files from Bucket" is disabled
-* `ui.defaultSourceBucket` - source bucket from `ui.sourceBuckets` that is selected by default; if it doesn't match any bucket then it's ignored
-* `ui.package_description` - a dictionary that maps package handle regular expressions or literals to JSONPath expressions of fields to show from package metadata in the package list view.
+Revise Package > Add files from Bucket; if the dictionary is not set
+or is empty the feature "Add files from Bucket" is disabled
+* `ui.defaultSourceBucket` - source bucket from `ui.sourceBuckets`
+that is selected by default; if it doesn't match any bucket then it's ignored
+* `ui.package_description` - a dictionary
```
Comment (sir-sigurd) [inline docs/Catalog/Preferences.md:58]: I think GitBook doesn't work well with multi-line list items: see 16aad503cffcc57d86b1145fb8df2ce897415773 and f152a237f85988713cfe6ade77c2ae93dd6a8783.
I'm not sure what useful markdownlint does besides that.

Comment (fiskus) [reply docs/Catalog/Preferences.md:58]: Already existed multiline list handled https://docs.quiltdata.com/catalog/preferences#properties
Maybe it was an old issue, that was fixed in GitBook?

Comment (fiskus) [reply docs/Catalog/Preferences.md:58]: linter is useful to not think about formatting anymore, and to not have unnecessary changes introduced by formatting

File: docs/api-reference/Bucket.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+<!-- markdownlint-disable -->
```
Comment (sir-sigurd) [inline docs/api-reference/Bucket.md:1]: docs/api-reference/*.md are generated from Python sources. Maybe these files could be ignored through markdownlint's config?

File: catalog/package.json [hunk 3]
```diff
@@ -210,6 +211,7 @@
     "jest": "^27.5.1",
     "jest-localstorage-mock": "^2.4.21",
     "lint-staged": "^12.4.1",
+    "markdownlint-cli": "^0.32.1",
```
Comment (sir-sigurd) [inline catalog/package.json:214]: Shouldn't we put it to the different place, since it lints docs for the whole project?

Comment (fiskus) [reply catalog/package.json:214]: I think we should, but don't know where. Probably, in `./testdocs` directory?

Comment (nl0) [reply catalog/package.json:214]: or you can just use `npx` and avoid having a `package.json` entirely

File: catalog/package-lock.json [hunk 4]
```diff
@@ -21237,13 +21237,13 @@
       }
     },
     "node_modules/micromatch": {
-      "version": "4.0.4",
-      "resolved": "https://registry.npmjs.org/micromatch/-/micromatch-4.0.4.tgz",
-      "integrity": "sha512-pRmzw/XUcwXGpD9aI9q/0XOwLNygjETJ8y0ao0wdqprrzDa4YnxLcz7fQRZr8voh8V10kGhABbNcHVk5wHgWwg==",
+      "version": "4.0.5",
```
Comment (sir-sigurd) [inline catalog/package-lock.json:21240]: Are changes in this file related to PR?

Comment (fiskus) [reply catalog/package-lock.json]: No, reverted, thanks

**Review by sir-sigurd — DISMISSED** [review]
I'd prefer to list ignored files in config, but that's up to you.

File: .github/workflows/js-ci.yml [hunk 5]
```diff
@@ -30,3 +30,16 @@ jobs:
...
+      run:
+        working-directory: docs
+    env:
+      NODE_OPTIONS: --max-old-space-size=4096
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.11'
+      - run: npx --package=markdownlint-cli markdownlint --ignore node_modules **/*.md
```
Comment (nl0) [inline .github/workflows/js-ci.yml:45]: i'd prefer to have this command accessible somewhere outside GH actions config, e.g. in a makefile or smth (even a standalone shell script would do), so it's easy to run it locally with the same options as in CI

**Review by nl0 — APPROVED** [review]

**Comment by sir-sigurd** [comment]
@fiskus 
is there anything to be done here?

**Comment by fiskus** [comment]
Yes. Something, according to your and Alexei feedback :)
I didn't decide what actually

**Comment by drernie** [comment]
I think these two suggestions (from above) are worth considering, to make it easier for people to incrementally adopt:

* list ignored files in config
* have this command accessible somewhere outside GH actions config, e.g. in a makefile or smth (even a standalone shell script would do), so it's easy to run it locally with the same options as in CI

Are those easy to do?

**Comment by fiskus** [comment]
Merged with master, resolved conflicts and fixed lint errors in new file

**Review by nl0 — APPROVED** [review]

File: .github/workflows/js-ci.yml (hunk 5 above)
Comment (fiskus) [reply .github/workflows/js-ci.yml:45]: Added task in Asana

**Comment by fiskus** [comment]
I ignored files in-place rather than in single config from my experience. It was convenient for me to chop one file by another:
1. remove first line
2. exec linter and see warnings
3. fix warnings, commit and merge (single file)

**Pull request merged**